memq-actor/src/main/java/io/appform/memq/mailbox/config/MailboxConfig.java

```diff
@@ -0,0 +1,23 @@
+package io.appform.memq.mailbox.config;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.appform.memq.mailbox.MailboxType;
+
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXISTING_PROPERTY, property = "type",
+        defaultImpl = UnBoundedMailboxConfig.class)
+@JsonSubTypes({
+        @JsonSubTypes.Type(name = "BOUNDED", value = BoundedMailboxConfig.class),
+        @JsonSubTypes.Type(name = "UNBOUNDED", value = UnBoundedMailboxConfig.class)
+})
+public abstract class MailboxConfig {
```

Comment:
- Add @ToString and @EqualsHashCode and @Getter.
- Make variable private final

Reply:
ideally this is not needed. Subclassing is not needed as well. You can just set size to Integer.MAX_VALUE by default and be done with it.